### ISSUE TITLE
api: add a stream endpoint that allows to query current state of the accounts and subscribe for updates

### ISSUE THREAD

**Issue by dshulyak** [spacemeshos/go-spacemesh#4646]
the closes example is StreamResults that streams transactions results.

endpoint should iterate over database (also see implementation in StreamResults) and emit account one by one to the stream. every account record stores a layer id, this is a logical timestamp when it was updated. for accounts we want to output only latest known state.
such stream should also support filtering by address, such that it won't look through whole database for single or couple of addresses.

stream should support watch=true, if set it will subscribe to in-memory events that are fired when account was updated after applying transactions.

right now every account event includes two states:
- current (what we store after consensus)
- projected (includes mempool state)
they are merged into single struct that is returned to the caller. i think that can be improved and modified to the following:

```go
type Account struct {
    State int // applied | projected
    Layer uint32 // logical timestamp when account was updated, if State is projected then it is irrelevant
    Address []byte
    Nonce uint64
    Balance uint64
    Template []byte // if empty account is not spawned. it is safe to submit transactions only after template is not nil and State is applied
}
```
